app/authentication/controllers/authentication_controller.ts

```diff
@@ -32,4 +36,60 @@ export default class AuthenticationController {
 
     return response.redirect().toRoute('home')
   }
+
+  async forgotPassword({ inertia }: HttpContext) {
+    return inertia.render('manager/authentication/forgot_password_page')
+  }
+
+  async forgotPasswordAction({ request, response }: HttpContext) {
+    // const resend = new Resend('resend')
+    const email = request.only(['email']).email
+
+    const userByEmail = await User.query().where('email', email).first()
+
+    if (userByEmail) {
+      const resetEmailUrl = router
+        .builder()
+        .prefixUrl('http://localhost:3333')
+        .params({ uid: userByEmail.uid })
+        .makeSigned('resetPassword', {
+          expiresIn: '1 hour',
+        })
+
+      await mail.send((message) => {
+        message
+          .to(email)
+          .from('alexbrgn.code@leadcode.fr')
+          .subject('Testing email form Adonis')
+          .text(resetEmailUrl)
+      })
+      return response.redirect().toRoute('home')
+    } else {
+      return response.response.statusCode === 404
+    }
+  }
+
+  async resetPassword({ request, response, inertia, params }: HttpContext) {
+    if (!request.hasValidSignature()) {
+      return response.badRequest('Invalid or expired URL')
+    }
+    const user = await User.findBy('uid', params.uid)
+
+    return inertia.render('manager/authentication/reset_password_page', { user })
+  }
+
+  async resetPasswordAction({ request, params }: HttpContext) {
+    if (request.input('password') === request.input('password_confirmation')) {
+      const data = await request.validateUsing(resetPasswordValidator)
+      const user = await User.findBy('uid', params.uid)
+
+      if (user) {
+        user.merge(data)
+        await user.save()
+        return { message: 'Changement de mot de passe effectué' }
```

Comment:
Don't return data or message because in our case, Inertia will wait an redirect instruction because frontend is delivery by backend side.

Use `redirect().back()` method